src/sfloader/fluid_sffile.c

```diff
@@ -1516,11 +1516,12 @@ static int load_pgen(SFData *sf, int size)
                 }
                 else
                 {
+                    SFZone * pzone = fluid_list_get(p2);
                     /* previous global zone exists, discard */
                     FLUID_LOG(FLUID_WARN, "Preset '%s': Discarding invalid global zone",
                               ((SFPreset *)(p->data))->name);
-                    *hz = fluid_list_remove(*hz, p2->data);
```

Comment:
I don't think this fix introduces unintended behaviour. As `*hz` is an alias for `p2`, you this line updates `p2` back to preset->zone, effectively starting the whole zone iteration from the beginning. The old code did not have that effect.